src/main/java/mat/server/CqlAttributesRemoteCall.java

```diff
@@ -0,0 +1,52 @@
+package mat.server;
+
+import mat.client.shared.CQLConstantContainer;
+import mat.client.shared.MatRuntimeException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestClientResponseException;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class CqlAttributesRemoteCall {
+
+    @Autowired
+    private RestTemplate restTemplate;
+
+    @Value("${mat.fhirMatServices.url}")
+    private String fhirMatServicesUrl;
+
+    private final static String FHIR_MAT_SERVICES_RECOURSE_FOR_ATTRIBUTES = "/find";
+
+    List<String> fhirDataTypeList = new ArrayList<>();
+    List<String> fhirAttributeList = new ArrayList<>();
+
+    public void getFhirAttributeAndDataTypes(CQLConstantContainer cqlConstantContainer) {
+
+        ResponseEntity<ConversionMapping[]> fhirResponse;
+        try {
+            fhirResponse = restTemplate.getForEntity(fhirMatServicesUrl + FHIR_MAT_SERVICES_RECOURSE_FOR_ATTRIBUTES, ConversionMapping[].class);
+        } catch (RestClientResponseException e) {
+            throw new MatRuntimeException(e);
+        }
+        ConversionMapping[] conversionMappings = fhirResponse.getBody();
+
+        for (ConversionMapping conversionMapping : conversionMappings) {
+
+            if(!fhirDataTypeList.contains(conversionMapping.getMatDataTypeDescription())) {
```

Comment:
Basically you are using an array list to and check contains operation in a loop. It's potentially an O(N^2) operation. 
Should you use a Set collection (HashSet or LinkedHashSet if order is important) instead?

Reply:
Also, if the data is not changed often we might want to cache attributed/data

Reply:
The reason I used List, is because the existing functionality will not change for adding Fhir Attributes.